c2corg_api/views/document.py

```diff
@@ -30,3 +31,64 @@ def _create_new_version(self, document):
         DBSession.add(meta_data)
         DBSession.add_all(versions)
         DBSession.flush()
+
+    def _update_version(self, document, comment, update_type, changed_langs):
+        assert update_type != UpdateType.NONE
+
+        meta_data = HistoryMetaData(comment=comment)
+        archive = self._get_document_archive(document, update_type)
+
+        cultures = \
+            self._get_cultures_to_update(document, update_type, changed_langs)
+        locale_versions = []
+        for culture in cultures:
+            locale = document.get_locale(culture)
+            locale_archive = self._get_locale_archive(locale, changed_langs)
+
+            version = DocumentVersion(
+                document_id=document.document_id,
+                culture=locale.culture,
+                version=999,
```

Comment:
> The requirements demand that the old URLs are kept in place.

I think that's mostly important for the documents main URL, for instance http://www.camptocamp.org/outings/681428/fr/le-luisin-de-vallorcine-a-salanfe

Ditto for documents lists (with or without parameters): http://www.camptocamp.org/outings/list/areas/14421/orderby/date/order/desc

For secondary pages such as history, diff, old version, etc. if it's not such a big deal if the URLs change.